flake.nix

```diff
@@ -0,0 +1,209 @@
+      pkgs.callPackage ./c/derivation.nix {
+        stdenv = if clangSupport then pkgs.clangStdenv else pkgs.stdenv;
+        inherit ltoSupport debugSupport;
+      };
+
+    mkCpp = {ltoSupport ? false, debugSupport ? false}:
+      pkgs.callPackage ./cpp/derivation.nix {
+        inherit gitignoreSource ltoSupport debugSupport;
+      };
+      
+    mkGo = {...}: pkgs.callPackage ./go/derivation.nix {
+      inherit gitignoreSource;
+      inherit (gomod2nix'.lib) buildGoApplication;
+      gomod2nix = gomod2nix'.gomod2nix;
+    };
+
+    mkNim = {debugSupport ? false, speedSupport ? false}:
+      pkgs.callPackage ./nim/derivation.nix {
+        inherit (pkgs.nimPackages) buildNimPackage;
+        inherit gitignoreSource nim-argparse debugSupport speedSupport;
+        inherit (pkgs.llvmPackages_14) bintools;
+      };
+
+    mkPhp = {opcacheSupport ? false}:
+      pkgs.callPackage ./php/derivation.nix {
+        inherit gitignoreSource php-sdl opcacheSupport;
+      };
+
+    mkPy = {mypycSupport ? false}:
+      pkgs.callPackage ./py/derivation.nix {
+        inherit mypycSupport gitignoreSource;
+        pythonPackages = pkgs.python310Packages;
+      };
+
+    mkRs = {ltoSupport ? false, debugSupport ? false}:
+      pkgs.callPackage ./rs/derivation.nix {
+        naersk = naersk';
+        inherit gitignoreSource ltoSupport debugSupport;
+      };
+
+    mkZig = {safeSupport ? false, fastSupport ? false}:
+      pkgs.callPackage ./zig/derivation.nix {
+        inherit zig zig-sdl zig-clap safeSupport fastSupport gitignoreSource;
+      };
+
+  in rec {
+    packages = rec {
+      inherit utils;
+      
+      c-debug = mkC { debugSupport = true; };
+      c-lto = mkC { ltoSupport = true; };
+      c-release = mkC { };
+      # c-clang-debug = mkC { debugSupport = true; clangSupport = true; };
+      # c-clang-lto = mkC { ltoSupport = true; clangSupport = true; };
+      # c-clang-release = mkC { clangSupport = true; };
+      c = hiPrio c-release;
+
+      cpp-release = mkCpp {};
+      cpp-debug = mkCpp { debugSupport = true; };
+      cpp-lto = mkCpp { ltoSupport = true; };
+      cpp = hiPrio cpp-release;
+      
+      go = mkGo {};
+
+      nim-release = mkNim {};
+      nim-debug = mkNim { debugSupport = true; };
+      nim-speed = mkNim { speedSupport = true; };
+      nim = hiPrio nim-release;
+
+      php-release = mkPhp {};
+      php-opcache = mkPhp { opcacheSupport = true; };
+      php = hiPrio php-release;
+
+      py = mkPy {};
+      # match statement support is only in myypc master
+      # https://github.com/python/mypy/commit/d5e96e381f72ad3fafaae8707b688b3da320587d
+      # mypyc = mkPy { mypycSupport = true; };
+      
+      rs-debug = mkRs { debugSupport = true; };
+      rs-release = mkRs { };
+      rs-lto = mkRs { ltoSupport = true; };
+      rs = hiPrio rs-release;
+      
+      zig-fast = mkZig { fastSupport = true; };
+      zig-safe = mkZig { safeSupport = true; };
+      zig = hiPrio zig-fast;
```

Comment:
@jbboehr I do think the changes above that aren't overlay related are still worthwhile; I can split https://github.com/rrbutani/rosettaboy/commit/e8711d9da1671a3d8fc6cfce9f81194c8fa2212b so that the overlay changes are separate from the `newScope` changes.